src/targets/anvil.js

```diff
@@ -27,46 +31,85 @@ export class Anvil extends ClickableObject {
         // Add invisible button for anvil
         this.sprite = new Button(scene, x, y, width, height);
         this.sprite.on("pointerdown", (pointer) => {
-            if (pointer.rightButtonDown() && !pointer.leftButtonDown()) {
+            if (pointer.rightButtonDown() && !pointer.leftButtonDown() && this.modalWindow.visible == false) {
                 this.createRightClickMenu(pointer.x, pointer.y, this.actions);
             } else {
                 this.clickTarget();
             }
         });
+
+        //Create modal window
+        this.modalWindow = new ModalWindow(this.scene);
     }
 
     async clickTarget() {
+        if (this.modalWindow.visible == true) {
+            return;
+        }
+
+        const currentRecipe = this.modalWindow.getChoice();
+
+        if (currentRecipe != "None") {
+            this.smith(currentRecipe);
+        }
+        else {
+            this.selectRecipe();
+        }
+    }
+
+    async hasMaterials() {
         const inv = this.scene.dashboard.inventory;
         const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
 
         // See if bar is selected
         const selectedIndex = inv.curSelectedItemIndex;
         if (selectedIndex < 0) {
             chat.writeText("Select a bar in your inventory first.");
-            return;
+            return false;
         }
 
         const hasHammer = inv.getInventoryIndex("Hammer");
 
         if (hasHammer < 0) {
             chat.writeText("You need a hammer to work the metal with.");
+            return false;
+        }
+
+        return true;
+    }
+
+    async selectRecipe(itemName) {
+        if (this.hasMaterials() == false) {
             return;
         }
 
+        const inv = this.scene.dashboard.inventory;
+        const selectedIndex = inv.curSelectedItemIndex;
+
         const selectedItem = inv.inventory[selectedIndex];
+        
+        let elements = [];
+        this.modalWindow.clearElements();
 
         switch (selectedItem.name) {
             case "Bronze Bar":
-                this.smith("BronzeDagger");
+                    elements = ["BronzeDagger", "BronzeSword", "BronzeScimitar"];
```

Comment:
I think I'm still in favor of having all of the modal window specifics inside of a child modal window class. i.e. bronze-smithing-modal-window.js or if all of the different smithing modals will be similar then maybe a smithing-modal-window that takes "bronze" as an argument. 

I prefer it that way for two reasons:
1) the modal window will be used for other things like the shop which, in addition to having different elements, could have a different background, multiple tabs, etc. Having to insert all that configurable info every time we create a modal window would be messy as opposed to just a single statement creating a child modal window class that has all the info inside.
2) the OOP model is so deeply ingrained in everything else we are doing it might as well continue

@CurtisGreen thoughts?

Reply:
I think your approach is best. I was overengineering a bit. Some future modal windows like shop and bank need to be a little customizable since the options can vary.

Reply:
Hard to say until we make a new modal and see what we need/don't need to be generic